sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/graph/GraphClient.java

```diff
@@ -51,43 +48,43 @@ public GraphClient(AadAuthenticationProperties properties, RestTemplateBuilder r
      */
     public GroupInformation getGroupInformation(String accessToken) {
         GroupInformation groupInformation = new GroupInformation();
-        final ObjectMapper objectMapper = JacksonObjectMapperFactory.getInstance();
         String aadMembershipRestUri = properties.getGraphMembershipUri();
         while (aadMembershipRestUri != null) {
-            Memberships memberships;
-            try {
-                String membershipsJson = getUserMemberships(accessToken, aadMembershipRestUri);
-                memberships = objectMapper.readValue(membershipsJson, Memberships.class);
-            } catch (IOException ioException) {
-                LOGGER.error("Can not get group information from graph server.", ioException);
-                break;
-            }
-            for (Membership membership : memberships.getValue()) {
-                if (isGroupObject(membership)) {
-                    groupInformation.getGroupsIds().add(membership.getObjectID());
-                    groupInformation.getGroupsNames().add(membership.getDisplayName());
+            Optional<Memberships> userMemberships = getUserMemberships(accessToken, aadMembershipRestUri);
+            if (userMemberships.isPresent()) {
+                for (Membership membership : userMemberships.get().getValue()) {
+                    if (isGroupObject(membership)) {
+                        groupInformation.getGroupsIds().add(membership.getObjectID());
+                        groupInformation.getGroupsNames().add(membership.getDisplayName());
+                    }
                 }
+                aadMembershipRestUri = userMemberships
+                    .map(Memberships::getOdataNextLink)
+                    .orElse(null);
+            } else {
+                break;
             }
-            aadMembershipRestUri = Optional.of(memberships)
-                                           .map(Memberships::getOdataNextLink)
-                                           .orElse(null);
         }
         return groupInformation;
     }
 
-    private String getUserMemberships(String accessToken, String urlString) throws IOException {
+    private Optional<Memberships> getUserMemberships(String accessToken, String urlString) {
```

Comment:
We could make this method package private and then test this method directly. So we don't need to capture the system.out in our test cases. Just verify it returns empty will be okay.